http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -124,13 +124,56 @@ final class LeoKubernetesServiceInterp[F[_]: Parallel](
         )
       )
 
+      lastUsedApp <- diskResultOpt.flatTraverse { diskResult =>
+        if (diskResult.creationNeeded) F.pure(none[LastUsedApp])
+        else {
+          (diskResult.disk.formattedBy, diskResult.disk.galaxyRestore) match {
```

Comment:
The check for `formattedBy` happens in `RuntimeServiceInterp.processPersistentDiskRequest` -- not sure it needs to be repeated here? Feels like this code could just match on `diskResult.disk.galaxyRestore` unless I'm missing something..

Reply:
there might be a bit overlapping, but not repeated really...`processPersistentDiskRequest` only cares about `formattedBy` and whether a disk is attached....here, we're checking between `formattedBy` and `galaxyRestore`...I don't want to mix `galaxyRestore` into `processPersistentDiskRequest` since it's used for other flow as well